test/TensorFlowRuntime/test_remote.swift

```diff
@@ -0,0 +1,33 @@
+// RUN: %target-run-simple-swift  | %FileCheck %s
+// REQUIRES: executable_test
+// REQUIRES: swift_test_mode_optimize
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import Darwin
+#else
+import Glibc
+#endif
+
+import TensorFlow
+import TensorFlowUnittest
+import StdlibUnittest
+import CTensorFlow
+
+var RemoteSessionTests = TestSuite("RemoteSessionTest")
+
+// Just a dummy test to trigger Tensor-related code paths in compiler.
+public func add(a: Tensor<Float>, b: Tensor<Float>) -> Tensor<Float> {
+  return a + b
+}
+
+RemoteSessionTests.testAllBackends("dummyAddTest") {
+  _RuntimeConfig.printsDebugLog  = true
+  expectNearlyEqualWithScalarTensor(3.0, add(a: Tensor<Float>(1.0), b: Tensor<Float>(2.0)))
+}
+
+// CHECK-LABEL: There are 4 devices.
```

Comment:
i'm debating whether it's a good idea to CHECK on the output of debug logs, which are meant for human consumption, and are subject to change.

I do like the content of the messages we want to CHECK here though. Perhaps one middle ground is to add comment to those specific `debugLog()` stmts, saying their output are being CHECK'ed, so do not update them.

Another option is to leave this as is, and revisit it in case this test gets accidentally broken down the road. You can decide.

Reply:
Also, consider abstracting out the CHECK content a bit. e.g. in a GPU build, there could be more than 4 devices in total.

how about we just CHECK on `task:0` and `task:1`, to confirm there are some remote workers?

Reply:
The number of devices is unpredictable as you observed. WIll try to abstract it. 

> how about we just CHECK on task:0 and task:1, to confirm there are some remote workers?

That is a good. Let me see if there is an API to check if something is remote.